pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -101,9 +104,16 @@ func newRepositoryWithClient(registryClient client.Interface, quotaClient kclien
 		}
 	}
 
-	nameParts := strings.SplitN(repo.Name(), "/", 2)
+	allowSchema2 := true
```

Comment:
I believe we want to default to `false` don't we? We don't want to break existing deployments with `docker-1.9`.